lib/dependabot/utils/java_script/requirement.rb

```diff
@@ -84,7 +84,27 @@ def convert_tilde_req(req_string)
 
         def convert_hyphen_req(req_string)
           lower_bound, upper_bound = req_string.split(/\s+-\s+/)
-          [">= #{lower_bound}", "<= #{upper_bound}"]
+          lower_bound_parts = lower_bound.split(".")
+          lower_bound_parts.fill("0", lower_bound_parts.length...3)
+
+          upper_bound_parts = upper_bound.split(".")
+          upper_bound_range = nil
+          if upper_bound_parts.length < 3
+            # When upper bound is a partial version treat these as an X-range
+            last_index = upper_bound_parts.length - 1
+            upper_bound_parts = upper_bound_parts.map.with_index do |part, i|
```

Comment:
This could just be:
```
if upper_bound_part[-1].to_i.positive?
  upper_bound_part[-1] = upper_bound_part[-1].to_i + 1 
end
```

No need for the loop, or the `last_index` variable.